crates/vm/levm/docs/faq.md

```diff
@@ -0,0 +1,45 @@
+On the other hand, the EVM specification requires all addresses to be in `U256`. Therefore, every opcode treats its arguments as `U256` values.
+The problem arises in the opcodes that need to acess a specific index on a data structure (e.g. `CALLDATA`, `CODECOPY`, `EXTCODECOPY`, etc).
+These operands receive offsets and indexes in `U256`, but the data structure they have to access (e.g. `Memory` or  `Calldata`) **require a `usize`**. Therefore, those paramenters need to be cast from `U256` to `usize`.
+The problem is, `U256`'s representation ranger is larger than `usize`'s; so not all numbers can be successfuly cast. In these cases, special attention is needed.
+
+The main way to deal with theses cases (at least, at the time of writing) is to **cast the value only when you know it can fit**. Before casting to `usize`, we compare the size of the index in `U256` with the length of the datastructure it wants to access. Here's an example from the `EXTCODECOPY` opcode (NOTE: the code snippet is a simplified/altered version to demonstrate this pattern. The actual implementation is fairly different):
+
+```rust
+///  bytecode: Represents the EVM bytecode array to be executed.
+0:   pub fn op_extcodecopy(bytecode_offset: U256, bytecode: Bytes, vector_size: usize) -> Result<(), Err> {
+
+        (...)
+
+1:       let mut data = vec![0u8; vector_size];
+2:
+3:       let bytecode_length: U256 = bytecode.len().into();
+4:       if bytecode_offset < bytecode_length {
+5:           let offset: usize = offset
+6:               .try_into()
+7:               .map_err(|_| InternalError::ConversionError)?;
+8:           // After this the data vector is modified
+
+        (...)
+
+9:      }
+10:     memory.store_data(&data);
+11: }
+```
+Some context: It is not important what this operand does. The only thing that matters for this example is that `EXTCODECOPY` stores a `data` vector in memory. The offset it receives will tell `EXTCODECOPY` which parts of the bytecode to skip, and which parts it will copy to memory. Skipped sections will be filled with 0's.
+
+- In line `1` we create the vector which we will return.
+- In line `3` we get the `bytecode` array length. Since `.len()` returns a `usize` we need to cast it to `U256`, in order to compare itwith `bytecode_offset`. Luckily, `usize` always fits into `U256`, so this will never fail.
+- In line `4` we check if the calldata offset is larger than the calldata itself. If this is the case, there's no data to copy. So we do not want to modify the vector.
+    -  Do note that, after this check we can safely cast the bytecode to `usize`. This is done in line `5`. This is because there is a limit to the contract's bytecode size. For more information, read [this article](https://ethereum.org/en/developers/docs/smart-contracts/#limitations).
+    -  We return an InternalError because line 5 should never fail. If it fails, then it means there's a problem with the VM itself.
```

Comment:
```suggestion
    -  We return an `InternalError` because line 5 should never fail. If it fails, then it means there's a problem with the VM itself.
```